### PR TITLE
Deployment tags

### PR DESCRIPTION
**Pull request by mikedecosta**
This creates a textarea to input plan tags for the deployments API. The tags should be input in JSON format as a list of lists for the disjunctive normal form. `(A && B && C) || (A && D) || ...`

Eventually would like to use a better input method, like adding individual select-areas via a [+] button to input 1 set of anded tags by selecting tags from a prepopulated filterable selection list.

For now this is what we get because of time.

bamboo input plan tags

This shouldn't be deployed until the deployment API is ready.

### CODE REVIEW

**Comment by hpsniper** [comment]
This is not actually ready, just had a chat with @johnkbrew and this is going to happen at the same time as some other changes to the deployment API so more UI updates need to happen.

**Comment by srvance** [comment]
Our tags are key-optional value pairs. How does this work with ones that have values?
`[{"name": "stage", "value": "dev"}, {"name": "default"}]`

File: src/main/resources/editCreateDeployment.ftl [hunk 1]
```diff
@@ -47,19 +101,39 @@ var mabl =
...
+            mabl.helpers.clearDropdown(dropdown, emptyMessage);
+            if(result !== "") {
+                AJS.$.each(result, function(k, v) {
+                    dropdown.append('<option value="' + v.uri + '">' + v.uri + '</option>');
+                });
+            }
+        },
+
+        showHide: function(field, value) {
+            if (value == "_mabl_input_custom") {
```
Comment (twistedpair) [inline src/main/resources/editCreateDeployment.ftl]: `===`

File: src/main/resources/editCreateDeployment.ftl [hunk 2]
```diff
@@ -47,19 +101,39 @@ var mabl =
...
+                    dropdown.append('<option value="' + v.name + '">' + v.name + '</option>');
                 });
             }
+        },
+
+        buildUriDropdown: function(result, dropdown, emptyMessage) {
+            mabl.helpers.clearDropdown(dropdown, emptyMessage);
+            if(result !== "") {
+                AJS.$.each(result, function(k, v) {
+                    dropdown.append('<option value="' + v.uri + '">' + v.uri + '</option>');
```
Comment (twistedpair) [inline src/main/resources/editCreateDeployment.ftl]: Does the Atlas/Bamboo framework do anything to escape outputs, so we don't introduce XSS threats into these build CI setups?

File: src/main/java/com/mabl/domain/GetDeploymentsResult.java [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+    @JsonCreator
+    public GetDeploymentsResult(
+            @JsonProperty("environments") final List<Deployment> deployments
+    ) {
+        this.deployments = deployments;
+    }
+
+    @SuppressWarnings("WeakerAccess")
+    public static class Deployment {
+        public final String id;
```
Comment (twistedpair) [inline src/main/java/com/mabl/domain/GetDeploymentsResult.java]: We should make sure the requisite documentation disclaims what we're tracking. Some customers might be perturbed, so we should be clear what we're using.

**Comment by hpsniper** [comment]
The changes to the api never made it live, so this won't be used at least for now

**Pull request closed without merging**